src/components/Dashboard/ThisWeek/NewsletterMenu/MenuItems/LogNewsLetter/LogNewsletter.tsx

```diff
@@ -31,28 +56,280 @@ const CloseButton = styled(IconButton)(({ theme }) => ({
   },
 }));
 
-const LogNewsletter = ({ handleClose }: Props): ReactElement<Props> => {
+const LogFormControl = styled(FormControl)(() => ({
+  width: '100%',
+}));
+
+const LogFormLabel = styled(FormLabel)(({ theme }) => ({
+  margin: theme.spacing(1, 0),
+  fontWeight: 'bold',
+  color: theme.palette.primary.dark,
+  '& span': {
+    color: theme.palette.error.main,
+  },
+}));
+
+const LogFormControlLabel = styled(FormControlLabel)(({ theme }) => ({
+  '& span.MuiFormControlLabel-label': {
+    fontWeight: 'bold',
+    color: theme.palette.primary.dark,
+  },
+}));
+
+const LogTextField = styled(TextField)(({ theme }) => ({
+  '& div': {
+    padding: theme.spacing(1),
+  },
+}));
+
+const taskSchema: yup.SchemaOf<
+  Omit<TaskCreateInput, 'result' | 'nextAction'>
+> = yup.object({
+  id: yup.string().nullable(),
+  activityType: yup.mixed<ActivityTypeEnum>(),
+  subject: yup.string().required(),
+  startAt: yup.string().nullable(),
+  completedAt: yup.string().nullable(),
+  tagList: yup.array().of(yup.string()).default([]),
+  contactIds: yup.array().of(yup.string()).default([]),
+  userId: yup.string().nullable(),
+  notificationTimeBefore: yup.number().nullable(),
+  notificationType: yup.mixed<NotificationTypeEnum>(),
+  notificationTimeUnit: yup.mixed<NotificationTimeUnitEnum>(),
+});
+
+const LogNewsletter = ({
+  accountListId,
+  handleClose,
+}: Props): ReactElement<Props> => {
   const { t } = useTranslation();
 
+  const [commentBody, changeCommentBody] = useState('');
+
+  const [createTask, { loading: creating }] = useCreateTaskMutation();
+  const [createTaskComment] = useCreateTaskCommentMutation();
+
+  const initialTask: TaskCreateInput = {
+    activityType: ActivityTypeEnum.NewsletterPhysical,
+    completedAt: null,
+    contactIds: [],
+    id: null,
+    nextAction: null,
+    notificationTimeBefore: null,
+    notificationTimeUnit: null,
+    notificationType: null,
+    result: null,
+    startAt: DateTime.local().minus({ hours: 1 }).startOf('hour').toISO(),
+    subject: '',
+    tagList: [],
+    userId: null,
+  };
+
+  const onSubmit = async (attributes: TaskCreateInput) => {
+    const body = commentBody.trim();
+    try {
+      await createTask({
+        variables: {
+          accountListId,
+          attributes,
+        },
+        update: (_cache, { data }) => {
+          if (data?.createTask?.task.id && body !== '') {
+            const id = uuidv4();
```

Comment:
This is required? Weird. Maybe we should get the API to generate that.

Reply:
Looks like it, I copied this from how comments are created currently: https://github.com/CruGlobal/mpdx-react/blob/9e0cc6c438902c4ccd06c1cb03a6e27da3f47c6d/src/components/Task/Drawer/CommentList/Form/Form.tsx#L61-L68